sound/soc/soc-pcm.c

```diff
@@ -882,6 +882,9 @@ static int soc_pcm_hw_clean(struct snd_pcm_substream *substream, int rollback)
 			snd_soc_dai_digital_mute(dai, 1, substream->stream);
 	}
 
+	/* run the stream event */
+	snd_soc_dapm_stream_stop(rtd, substream->stream);
+
```

Comment:
I don't get why you call this here? I would have done it after the set of hw_free() below. if you turn the power off before actually de-preparing the resources, who know what might happen.

Would moving this to line 901 make sense?

Reply:
@plbossart the startup sequence is hw_params -> DAPM_SREAM_START. So the opposite should be DAPM_STREAM_STOP and then hw_free no? 

This change is particularly important for my other PR https://github.com/thesofproject/linux/pull/2371.

I set up the pipelines/widgets in hw_params and free them in hw_free. So ideally all DAPM power events should occur between the two to keep them balanced.

Reply:
You may be right. I always thought hw_free was the dual of prepare, but the ALSA documentation says it's the dual of hw_params. Maybe it's the SoundWire logic that twists my brain, or what's left of it.

Reply:
If you are good with this change, can I post this to alsa-devel @plbossart ?

Reply:
let's check with @lgirdwood this piece of code is 8 years old 

````
01d7584cd2e5a (Liam Girdwood            2012-04-25 12:12:49 +0100 1886)         /* run the stream event for each BE */
````

Liam, any reason why the DAPM events were handled in close instead of hw_free?

Reply:
@lgirdwood ping?

Reply:
@plbossart probably related to support for OMAP4 DSP that never materialsed. I would say it's safe to move.

Reply:
Thanks @lgirdwood can you approve so that we have your Reviewed-by tag automagically?

Reply:
@lgirdwood ping